✅ Started W&B orchestrator run: orchestrator_20250620_183955
✅ Set MLflow experiment to: 'MLOps-Group-Project-Experiment'
▶️  Running step: 'data_load'...
2025/06/20 18:39:57 INFO mlflow.utils.virtualenv: Installing python 3.13.1 if it does not exist
Error executing job with overrides: []
Traceback (most recent call last):
  File "/Users/aviv/Library/Mobile Documents/com~apple~CloudDocs/study/MLOps_Group_Project/main.py", line 117, in main
    mlflow.run(step_dir, "main", parameters=params)
    ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/mlflow/projects/__init__.py", line 357, in run
    submitted_run_obj = _run(
        uri=uri,
    ...<12 lines>...
        docker_auth=docker_auth,
    )
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/mlflow/projects/__init__.py", line 110, in _run
    submitted_run = backend.run(
        uri,
    ...<5 lines>...
        experiment_id,
    )
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/mlflow/projects/backend/local.py", line 163, in run
    activate_cmd = _create_virtualenv(
        local_model_path=work_dir_path,
    ...<4 lines>...
        extra_env=env_vars,
    )
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/mlflow/utils/virtualenv.py", line 265, in _create_virtualenv
    python_bin_path = _install_python(
        python_env.python, pyenv_root=pyenv_root_dir, capture_output=capture_output
    )
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/mlflow/utils/virtualenv.py", line 138, in _install_python
    _exec_cmd(
    ~~~~~~~~~^
        [pyenv_bin_path, "install", *pyenv_install_options, version],
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<3 lines>...
        extra_env=extra_env,
        ^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/mlflow/utils/process.py", line 119, in _exec_cmd
    process = subprocess.Popen(
        cmd,
    ...<2 lines>...
        **kwargs,
    )
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py", line 1036, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                        pass_fds, cwd, env,
                        ^^^^^^^^^^^^^^^^^^^
    ...<5 lines>...
                        gid, gids, uid, umask,
                        ^^^^^^^^^^^^^^^^^^^^^^
                        start_new_session, process_group)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py", line 1966, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'None'

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
